mvcc/backend/batch_tx.go

```diff
@@ -307,6 +303,17 @@ func (t *batchTxBuffered) unsafeCommit(stop bool) {
 	}
 }
 
+func waitAndRollback(tx *bolt.Tx, wg *sync.WaitGroup, lg *zap.Logger) {
```

Comment:
Declare as `func (t readTx) waitAndRollback(lg *zap.Logger) ` and call as `go t.backend.readTx.waitAndRollback(t.backend.lg)`?  Might even be reasonable to wire in a logger so it doesn't need to be passed in here?

Reply:
Unfortunately this won't work. readTx.tx will be pointing to a new bolt Tx immediately after this call.